R/spark_compile.R

```diff
@@ -239,25 +241,29 @@ spark_default_compilation_spec <- function(pkg = infer_active_package_name()) {
       spark_version = "1.5.2",
       scalac_path = find_scalac("2.10"),
       jar_name = sprintf("%s-1.5-2.10.jar", pkg),
-      jar_path = find_jar()
+      jar_path = find_jar(),
+      scala_filter = make_version_filter("1.5.2")
     ),
     spark_compilation_spec(
       spark_version = "1.6.1",
       scalac_path = find_scalac("2.10"),
       jar_name = sprintf("%s-1.6-2.10.jar", pkg),
-      jar_path = find_jar()
+      jar_path = find_jar(),
+      scala_filter = make_version_filter("1.6.1")
```

Comment:
@javierluraschi `scala_filter` is indeed used, otherwise the Scala compiler would complain.